src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/FixAllCodeRefactoringSuggestedAction.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.UnifiedSuggestions.UnifiedSuggestedActions;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
+{
+    /// <summary>
+    /// Suggested action for fix all occurrences for a code refactoring.  Note: this is only used
+    /// as a 'flavor' inside CodeRefactoringSuggestionAction.
+    /// </summary>
+    internal sealed class FixAllCodeRefactoringSuggestedAction : SuggestedAction, IFixAllCodeRefactoringSuggestedAction
+    {
+        /// <summary>
+        /// The original code-action that we are a fix-all for.  This suggestion action
+        /// and our <see cref="SuggestedAction.CodeAction"/> is the actual action that 
+        /// will perform the fix in the appropriate document/project/solution scope.
+        /// </summary>
+        public CodeAction OriginalCodeAction { get; }
+
+        public FixAllState FixAllState { get; }
+
+        internal FixAllCodeRefactoringSuggestedAction(
+            IThreadingContext threadingContext,
+            SuggestedActionsSourceProvider sourceProvider,
+            Workspace workspace,
+            ITextBuffer subjectBuffer,
+            FixAllState fixAllState,
+            CodeAction originalCodeAction)
+            : base(threadingContext, sourceProvider, workspace, subjectBuffer,
+                   fixAllState.FixAllProvider, new FixAllCodeRefactoringCodeAction(fixAllState))
+        {
+            OriginalCodeAction = originalCodeAction;
+            FixAllState = fixAllState;
+        }
+
+        public override bool TryGetTelemetryId(out Guid telemetryId)
+        {
+            // We get the telemetry id for the original code action we are fixing,
+            // not the special 'FixAllCodeAction'.
+            telemetryId = OriginalCodeAction.GetTelemetryId(FixAllState.FixAllScope);
+            return true;
+        }
+
+        protected override async Task InnerInvokeAsync(
+            IProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
confused by this.  how does thsi help at all.  we switched to the UI thread, just to log something, then called base.InnerInvokeAsync?

Reply:
Just a clone of https://github.com/dotnet/roslyn/blob/e1629052738683dac234a4d75bb1d1c442c46d7d/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/FixAllSuggestedAction.cs#L68-L77

Will see if the code can be shared.